pkg/clustermanager/cluster_manager.go

```diff
@@ -709,11 +710,13 @@ func (c *ClusterManager) waitForControlPlaneReplicasReady(ctx context.Context, m
 
 func (c *ClusterManager) waitForMachineDeploymentReplicasReady(ctx context.Context, managementCluster *types.Cluster, clusterSpec *cluster.Spec) error {
 	var machineDeploymentReplicasCount int
+	readyReplicas, totalReplicas := 0, 0
 	for _, workerNodeGroupConfiguration := range clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations {
 		machineDeploymentReplicasCount += workerNodeGroupConfiguration.Count
 	}
 
 	isMdReady := func() error {
+		readyReplicas, totalReplicas, _ = c.countReplicasReady(ctx)
 		return c.clusterClient.ValidateWorkerNodes(ctx, clusterSpec.Cluster.Name, managementCluster.KubeconfigFile)
 	}
 
```

Comment:
Would you mind deleting l.723-726. The retry loop will execute N retries and if it succeeds on the first attempt the function returns as expected. As is, we have N+1 attempts which feels a little unnecessary.